### PR TITLE
Introduce Style Attribute

### PR DESCRIPTION
**Pull request by fdietze**
Hi, we introduced a Style attribute which uses the Style Module of Snabbdom. Is this going in the right direction? Should we make it usable like ScalaTags?

### CODE REVIEW

**Comment by fdietze** [comment]
How did you port the MDN data for the tags and attributes? Is it possible to do the same for Styles?

File: src/main/scala/snabbdom/VDomProxy.scala [hunk 1]
```diff
@@ -14,15 +14,18 @@ object VDomProxy {
...
-  def attrsToSnabbDom(attributes: Seq[Attribute]): (js.Dictionary[String], js.Dictionary[String]) = {
-    val (attrs, props) = attributes.partition {
-      case (a: Attr) => true
-      case (p: Prop) => false
+  def attrsToSnabbDom(attributes: Seq[Attribute]): (js.Dictionary[String], js.Dictionary[String], js.Dictionary[String]) = {
+    val attrsDict = js.Dictionary[String]()
+    val propsDict = js.Dictionary[String]()
+    val styleDict = js.Dictionary[String]()
+
+    attributes.foreach {
```
Comment (LukaJCB) [inline src/main/scala/snabbdom/VDomProxy.scala]: Can you change this to a fold? :)

Comment (fdietze) [reply src/main/scala/snabbdom/VDomProxy.scala]: I really thought about doing this with a fold. But I didn't come up with a way that has the same performance characteristics as my current solution. With building up a triple of lists inside a fold and constructing dictionaries afterwards, we have to iterate twice over every element. We cannot build an immutable triple of dictionaries. And in the end we have to write it into a mutable dictionary anyways: https://github.com/scala-js/scala-js/blob/master/library/src/main/scala/scala/scalajs/js/Dictionary.scala#L59

So I think this is a trade-off between being pseudo-pure and performance. I chose performance. Do you still prefer a fold? ;)

**Comment by PerWiklander** [comment]
> fdietze commented 3 days ago:
> How did you port the MDN data for the tags and attributes? Is it possible to do the same for Styles?

I copied most of it from ScalaTags and looked the rest up in MDN. I believe @lihaoyi had a script to do the bulk of the extraction in the beginning.

**Pull request merged**